tests/integration/telemetry/stats/prometheus/testdata/metadata_exchange_filter.yaml

```diff
@@ -15,11 +15,14 @@ spec:
         operation: INSERT_BEFORE
         value:
           name: envoy.filters.http.wasm
-          config:
-            config:
-              configuration: envoy.wasm.metadata_exchange
-              vm_config:
-                runtime: envoy.wasm.runtime.null
-                code:
-                  local:
-                    inline_string: envoy.wasm.metadata_exchange
+          typed_config:
```

Comment:
should this be
```
config:
  typed_config:
     ...
```

error log:
```
Exiting due to setup failure: exit status 1: Error from server: error when creating "/logs/artifacts/stats-filter-test-a94367dbd0d14/_suite_context/galley-workdir-713754910/networking_istio_io_EnvoyFilter_istio-bookinfo-1-32783_stats-filter-3.yaml": admission webhook "pilot.validation.istio.io" denied the request: configuration is invalid: 3 errors occurred:
	* Envoy filter: unknown message type "udpa.type.v1.TypedStruct"
	* Envoy filter: unknown message type "udpa.type.v1.TypedStruct"
	* Envoy filter: unknown message type "udpa.type.v1.TypedStruct"
```

Reply:
no, it should be typed_config at the top level. the error is due to galley not linking some UDPA protos. trying to move static linking into shared code.

Reply:
OMFG. I think this is again gogoproto vs goproto madness. I'll see if I can workaround it or not.